packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -742,3 +635,291 @@ test.serial('writeGranules() stores error on granule if any file fails', async (
   t.is(pgGranule.error.Error, 'Failed writing files to PostgreSQL.');
   t.true(pgGranule.error.Cause.includes('AggregateError'));
 });
+
+test.serial('writeGranuleFromApi() throws for a granule with no granuleId provided', async (t) => {
+  const {
+    knex,
+    granule,
+  } = t.context;
+
+  await t.throwsAsync(
+    writeGranuleFromApi({ ...granule, granuleId: undefined }, knex),
+    { message: 'Could not create granule record, invalid granuleId: undefined' }
+  );
+});
+
+test.serial('writeGranuleFromApi() throws for a granule with an invalid collectionId', async (t) => {
+  const {
+    knex,
+    granule,
+  } = t.context;
+
+  await t.throwsAsync(
+    writeGranuleFromApi({ ...granule, collectionId: 'wrong___collection' }, knex),
+    { message: 'Record in collections with identifiers {"name":"wrong","version":"collection"} does not exist.' }
+  );
+});
+
+test.serial('writeGranuleFromApi() throws for a granule with no collectionId provided', async (t) => {
+  const {
+    knex,
+    granule,
+  } = t.context;
+
+  await t.throwsAsync(
+    writeGranuleFromApi({ ...granule, collectionId: undefined }, knex),
+    { message: 'invalid collectionId: undefined' }
+  );
+});
+
+test.serial('writeGranuleFromApi() throws for a granule with an invalid collectionId provided', async (t) => {
+  const {
+    knex,
+    granule,
+  } = t.context;
+  const badCollectionId = `collectionId${cryptoRandomString({ length: 5 })}`;
+  await t.throwsAsync(
+    writeGranuleFromApi({ ...granule, collectionId: badCollectionId }, knex),
+    { message: `invalid collectionId: ${badCollectionId}` }
+  );
+});
+
+test.serial('writeGranuleFromApi() writes a granule to PostgreSQL and DynamoDB.', async (t) => {
+  const {
+    knex,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+  } = t.context;
+
+  const result = await writeGranuleFromApi({ ...granule }, knex);
+
+  t.is(result, `Wrote Granule ${granuleId}`);
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(knex, { granule_id: granuleId }));
```

Comment:
Everywhere we are doing a granulePgModel.get or exists, we should probably get in the habit of doing it like this (which you are doing below)

```
  const granuleRecord = await granulePgModel.get(
    knex,
    {
      granule_id: granuleId,
      collection_cumulus_id: collectionCumulusId,
    }
  );
```

The reason why is that in PG, only the combo of granule_id + collection_cumulus_id makes a granule unique, so otherwise the `get` or `exists` could return true/truthy for the `granule_id` if we had bad test setup even when our logic didn't create the record

Reply:
yup have to now that it has guards. d65a289996, and lint fix to go with a6322f9a88